File: android/aq-android-ben02/src/app/src/main/java/com/ostorlab/your/song/SettingsActivity.java
package com.ostorlab.your.song;

import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SwitchCompat;
import androidx.preference.PreferenceManager;

public class SettingsActivity extends AppCompatActivity {

    private SwitchCompat notificationsSwitch;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        notificationsSwitch = findViewById(R.id.notifications_switch);

        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);

        // Load saved preference
        boolean notificationsEnabled = sharedPreferences.getBoolean("notifications_enabled", false);
        notificationsSwitch.setChecked(notificationsEnabled);

        notificationsSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
            sharedPreferences.edit().putBoolean("notifications_enabled", isChecked).apply();
        });
    }
}

File: android/aq-android-ben18/src/app/src/main/java/co/ostorlab/myapplication/SettingsActivity.kt
package co.ostorlab.myapplication

import android.os.Bundle
import android.widget.Switch
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class SettingsActivity : AppCompatActivity() {
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        
        setupSettingsOptions()
    }
    
    private fun setupSettingsOptions() {
        findViewById<TextView>(R.id.tvSettingsTitle).text = "App Settings"
        
        val notificationSwitch = findViewById<Switch>(R.id.switchNotifications)
        val darkModeSwitch = findViewById<Switch>(R.id.switchDarkMode)
        val autoRefreshSwitch = findViewById<Switch>(R.id.switchAutoRefresh)
        
        // Set default values
        notificationSwitch.isChecked = true
        darkModeSwitch.isChecked = false
        autoRefreshSwitch.isChecked = true
    }
}
